### ISSUE TITLE
Granularity of XD- and XC-type errors

### ISSUE THREAD

**Issue by xml-project** [xproc/3.0-specification#342]
Related to issue #341 a more general question comes up to me: Since we have now multiple p:catch to handle different types of errors, does it still make sense to have such molecular errors as XD0011:

> It is a dynamic error if the resource referenced by a p:document element does not exist, cannot be accessed, or has an XML content type and is not a well-formed XML document.

Authors might want to have different error mechanisms for non existing documents, not accessible (protected) document or not well-formed documents. The my understanding such a molecular type of error was ok in XProc 1.0 were the basic audience for error message was human. In XProc 3.0 pipelines will consume errors very much, so I think we need more specific codes to make this possible.

One could argue that the pipeline should inspect the error message (in contract to the error code), but this is the path to non interoperable pipelines because different processors will display different error texts and users might use the pipeline in different language environment and want their error message e.h. in spanish.

**Comment by ndw** [comment]
I agree, we want explicit codes for distinct errors to the largest extent practical.

**Comment by gimsieke** [comment]
also related to #333

**Comment by ndw** [comment]
I don't think there's any specific action that we can take to close this issue. I've attempted to close #333 and I propose that we simply close this one. We all agree that more granularity is better in error codes.

**Comment by xml-project** [comment]
Not sure I want to close this. Although there is currently no specific action yet, I think we should keep it as a reminder to look over all XDs and XCs as one of the last editorial steps. Proposal: Change labels to "editorial".

**Comment by ndw** [comment]
🤷‍♂️ Ok.

**Comment by xml-project** [comment]
Am I the only one who cares about the question whether our error messages enable pipeline authors to catch them and do something meaningful with them for pipeline users?
If so: 🤷‍♂️Fell free to close this issue.

**Comment by ndw** [comment]
Of course you aren't. If having this issue hanging about open is helping, we can leave it open. I feel like we're doing better on that score because we've decided to care. Having this issue hanging around isn't helping *me* but that's fine.

**Comment by xml-project** [comment]
I thought of this issue as a reminder to have a final look at the XC???? and XD???? error. Since we are not finished with step library, I think it is still relevant. But may be we should move it to step issues.
